### PR TITLE
HOSTEDCP-2177: [CPO Refactor] Refactor cluster-version-operator

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/v2/cvo/ibm/preparepayload.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package ibm
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/v2/cvo/ibm/preparepayload.go:1]: why is this called ibm?

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/v2/cvo/ibm/preparepayload.go:1]: this is only called for ibm/powervs platform
https://github.com/openshift/hypershift/pull/5125/files#diff-213db240d2bbec8b38a0bc813f1683a1d16a6790e3e905038c574db9a9e5321dR26-R30

**Comment by muraee** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

**Pull request merged**